### PR TITLE
fix(flex): in breakpointMods set breakpoint as optional and add enums

### LINKED ISSUE #3244: PF4 Layout Flex: Typing improvements for breakpointMods
There is an inconsistency between documentation and typings for `breakpointMods`.
Typing for breakpointMods suggests `justify-content-flex-space-between` for `modifier` which is invalid and undefined. Correct value is `justify-content-space-between`.
Using `justify-content-space-between` leads to TypeError: 
```
TS2322: Type 'string' is not assignable to type '"column" | "flex" | "inline-flex" | "column-reverse" | "row" | "row-reverse" | "nowrap" | "wrap" | "wrap-reverse" | "full-width" | "spacer-none" | "spacer-xs" | "spacer-sm" | ... 44 more ... | "align-content-space-around"'.
```
`breakpoint` should be optional because docs shows `breakpoint` not being used with `justify-content-space-between`.
Documentation for PF4 show the correct implementation for this: https://patternfly-react.surge.sh/patternfly-4/documentation/react/layouts/flex#justify-content-space-between

`<Flex breakpointMods={[{ modifier: 'justify-content-space-between' }]} />`

### LINKED ISSUE #3248: Flex component TypeScript error
Attempting to use this simple Flex component example

```
<Flex className="example-border">
  <FlexItem breakpointMods={[{modifier: "align-right"}]}>Flex item</FlexItem>
  <FlexItem>Flex item</FlexItem>
</Flex>
```

And encountered the TypeScript error below:

Screen Shot 2019-10-30 at 5 45 45 PM

Also tried importing FlexBreakpointMod, but it is not exported properly.

Screen Shot 2019-10-30 at 5 45 15 PM

### PR DESCRIPTION
**Pull request by boaz0**
**What**:

closes #3244 
closes #3248

- Set breakpoint as optional in breakpointMd
- Add FlexModifiers and FlexItemModifiers
- Update examples

### CODE REVIEW

**Comment by rebeccaalpert** [comment]
@tlabaj wasn't there a discussion about how we wouldn't use enums in TypeScript components? Has that changed?

**Comment by boaz0** [comment]
@rebeccaalpert I didn't know that you're not going to use enums.
What should I use instead? Because if I am using literal types, passing a string will show a typescript error unless I will use the `as` keyword.

**Comment by rebeccaalpert** [comment]
@boaz0 - It sounds like enums are ok. (Wanted to check with more active PatternFly devs.) Back when we were converting all the components to TypeScript there had been some back-and-forth over declaring everything as strings vs. using enums.

File: packages/patternfly-4/react-core/src/layouts/Flex/FlexUtils.tsx [hunk 1]
```diff
@@ -2,98 +2,112 @@ import * as React from 'react';
...
+  'align-right' = 'align-right',
+  'align-left' = 'align-left',
+  'align-self-flex-start' = 'align-self-flex-start',
+  'align-self-flex-end' = 'align-self-flex-end',
+  'align-self-flex-center' = 'align-self-flex-center',
+  'align-self-flex-baseline' = 'align-self-flex-baseline',
+  'align-self-flex-stretch' = 'align-self-flex-stretch',
+  'justify-content-flex-end' = 'justify-content-flex-end',
+  'justify-content-center' = 'justify-content-center',
+  'justify-content-flex-space-between' = 'justify-content-flex-space-between',
```
Comment (sahil143) [inline packages/patternfly-4/react-core/src/layouts/Flex/FlexUtils.tsx]: @tlabaj Not sure if this value is correct. Docs example says to use `justify-content-space-between` https://github.com/patternfly/patternfly-react/pull/3258/files#diff-fee6334c94230e770989b535c35f87f7R428
Also, `justify-content-flex-space-between` doesn't work when this is used an undefined class is applied to the element.

Comment (boaz0) [reply packages/patternfly-4/react-core/src/layouts/Flex/FlexUtils.tsx]: :fearful: ho snap! I see this now. It looks like the Modifier type was in correct to begin with. It didn't have `justify-content-space-between` and the flex.css doesn't have `justify-content-space-between` but `justify-content-space-between`.

My mistake for not paying attention. I will suggest a fix

**Pull request merged**